commons/signservice-commons/src/main/java/se/idsec/signservice/security/sign/SignatureValidationResult.java

```diff
@@ -106,4 +107,28 @@
    */
   CertificateValidationResult getCertificateValidationResult();
 
+  /**
+   * Gets the URI identifier of the signature algorithm.
+   *
+   * @return signature algorithm URI identifier
+   */
+  String getSignatureAlgorithm();
+
+  /**
+   * Gets the claimed signing time.
+   * <p>
+   * This is a signing time asserted within the signature, not asserted by any external time stamp service.
+   * </p>
+   *
+   * @return the signing time (as millis since epoch)
+   */
+  Date getClaimedSigningTime();
```

Comment:
Or long (millis since epoch).

Reply:
Keeping Date to not change all implementations of this.